x/slashing/keeper.go

```diff
@@ -120,3 +130,36 @@ func (k Keeper) handleValidatorSignature(ctx sdk.Context, pubkey crypto.PubKey,
 	// Set the updated signing info
 	k.setValidatorSigningInfo(ctx, address, signInfo)
 }
+
+// AddValidators adds the validators to the keepers validator addr to pubkey mapping.
+func (k Keeper) AddValidators(vals []abci.Validator) {
+	for i := 0; i < len(vals); i++ {
+		val := vals[i]
+		pubkey, err := tmtypes.PB2TM.PubKey(val.PubKey)
+		if err != nil {
+			continue
+		}
+		k.addPubkey(pubkey, false)
+	}
+}
+
+// TODO: Make a method to remove the pubkey from the map when a validator is unbonded.
```

Comment:
I think this is pretty trivial, no? Why not just add the call to `Keeper#unbondValidator`?

Reply:
Given the discussion in https://github.com/tendermint/tendermint/pull/2177, we may just want to not do this at all, so no point figuring it out. (We can live with one hashmap entry per every validator on the next testnet)

Reply:
I see I see.